### ISSUE TITLE
Mimic file-structure with links

### ISSUE THREAD

**Issue by tdegeus** [tdegeus/shelephant#114]
# Proposal 1

`shelephant_hostinfo.yaml`:
```yaml
root: 
- ../mycontainer  # primary source
- /path/to/usb  # secondary source
- /path/to/network/mount  # 3rd source 
files:
- a.txt
- mydir/b.txt
```
**Todo:** to write this file the following should be possible
```yaml
shelphant_hostinfo --root ../container --root /path/to/usb --root /path/to/network/mount
```
by which the command merges the available
```yaml
../container/shelephant_dump.yaml
/path/to/usb/shelephant_dump.yaml
/path/to/network/mount/shelephant_dump.yaml
```
**Todo:** Think of protocol for a different file than `shelephant_dump.yaml`

Then 
```
shelephant_symlink
```
should create 
```
a.txt -> ../mycontainer/a.txt
mydir/b.txt -> /path/to/usb/b.txt
```
depending on the availability

* If none of the sources are available the function should create a dangling symlink, or remove an old symlink. **Todo:** it is not entirely clear what path the dangling symlink should have. 

* If a file with that name already exists, the function could:
  - throw
  - copy the file to one of the available sources: `shelephant_symlink --mv auto` or `shelephant_symlink --mv 0` or `shelephant_symlink --mv ../mycontainer` (path should match `shelephant_hostinfo.yaml`).

* An option `--prune` could be added to remove any symlinks that point to any of the roots.

## Plain file

The same should be possible with
`shelephant_dump.yaml`:
```yaml
- a.txt
- mydir/b.txt
```
and 
```
shelephant_symlink --root ../container
```
to have multiple sources one could have 
```
shelephant_symlink --root ../container --root /path/to/usb --root /path/to/network/mount
```
For `shelephant_hostinfo.yaml` input this function could:
* overwrite the source file
* be forced to have `--key` input

**Comment by tdegeus** [comment]
What would be a nice addition is
```yaml
root: 
- ../mycontainer  # primary source
- /path/to/usb  # secondary source
- /path/to/network/mount  # 3e source 
files:
- a.txt
- mydir/b.txt

**Comment by tdegeus** [comment]
This should be complemented with a command
```bash
shelephant_whereis
```
that should look in
```bash
../mycontainer/shelephant_dump.yaml
/path/to/usb/shelephant_dump.yaml
/path/to/network/mount/shelephant_dump.yaml
```
and have as output
```
- a.txt [cluster
- mydir/b.txt
```

**Comment by tdegeus** [comment]
# Proposal 2

Instead of having one `shelephant_hostinfo.yaml` with three alternative sources one can also have a several `shelephant_hostinfo.yaml` that are combined on the spot 

In this scenario you could have a dataset with several files;

```bath
sourceinfo_laptop.yaml
sourceinfo_usb.yaml
sourceinfo_cluster.yaml
```

Each of the files could have
```yaml
root: /my/path/on/the/system # can be relative on a local system
search: # information on how to auto-update the list of files
- command:
  - find . -iname '*.txt'
- ignore:
  - bak.*
# allow mount command?
files: # available files (stored in case the resource is not available)
- path: a.txt
  sha256: ...
- path: mydir/b.txt
  sha256: ...
```

The usage would be any of the following:
```bash
shelephant_symlink # using `sourceinfo.yaml`
shelephant_symlink laptop usb cluster
shelephant_symlink sourceinfo_laptop.yaml sourceinfo_usb.yaml sourceinfo_cluster.yaml
```
whereby `sourceinfo.yaml` would list the priorities
```yaml
- sourceinfo_laptop.yaml
- sourceinfo_usb.yaml
- sourceinfo_cluster.yaml
```

Creating `sourceinfo_mysource.yaml` could be done as follows:

```bash
cd /path/to/mysource
shelephant_dump ... [--sha256] [--date] [--size]
cd /path/to/my/dataset
shelephant_hostinfo -p /path/to/source -f sourceinfo_mysource.yaml
```

and later

```bash
shelephant_update sourceinfo_mysource.yaml [--sha256] [--date] [--size] # with option to add information, existing information is updated
```

The big advantage is that:
- creating such files involves no new tools (only a small extension)
- such files can be updated very easily: **todo** think of a protocol to do that on the remote
- finding the difference between storage locations (below) does not involve any new operations
- this files can also be used by a `shelephant_status myhost.yaml` to check 'unmanaged' files (can be combined with tool to find the difference between storage locations)

**Comment by tdegeus** [comment]
Available from v0.22
